hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -99,6 +100,7 @@ public LocalRegionCache(final String name, final HazelcastInstance hazelcastInst
         } else {
             topic = null;
         }
+        hazelcastInstance.getCluster().addMembershipListener(this);
```

Comment:
I'm not sure it is a problem.  I don't think the Hibernate cache regions are destroyed.